File: string/basic/function-copy-string-iterative-recursive/copy-string.js
const copyString = (string1, string2) => {
  if (!string1) {
    return string2;
  }

  string2 = string2 + string1[0];
  string1 = string1.slice(1);

  return copyString(string1, string2);
};

console.log(copyString("hello there how are you", ""));

File: string/basic/function-copy-string-iterative-recursive/copy-string.py
def copy_string(string1, string2):
    if string1 == '':
        return string2

    string2 = string2 + string1[0]
    string1 = string1[1:]

    return copy_string(string1, string2)

print(copy_string('hello there whats up', ''))
